src/cursor.rs

```diff
@@ -251,3 +253,121 @@ impl<'a> Iterator for TailingCursor<'a> {
         }
     }
 }
+
+type DocArray = VecDeque<Document>;
+type CursorId = i64;
+
+pub struct BatchCursor<'a> {
+    cursor:     Cursor<'a>,
+    db:         &'a Database<'a>,
+    coll_name:  String,
+    cursor_id:  Option<CursorId>,
+    documents:  Option<DocArray>
+
+}
+
+impl<'a> BatchCursor<'a> {
+    pub fn new(
+        cursor: Cursor<'a>,
+        db: &'a Database<'a>,
+        coll_name: String
+    ) -> BatchCursor<'a> {
+        BatchCursor {
+            cursor,
+            db,
+            coll_name,
+            cursor_id: None,
+            documents: None
+        }
+    }
+
+    fn get_cursor_next(&mut self) -> Option<Result<Document>> {
```

Comment:
have you considered implementing `Iter` for this type? Then you can integrate with everything.

Reply:
I think I have this below (line 347)

Reply:
Sorry, missed that these are private. Ignore what I said.